sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,425 @@
+  // after waking up from the deep sleep mode.
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+  // Total number of iterations for each test phase
+  rnd_num_iterations = flash_ctrl_testutils_counter_get(kCounterTestPhase);
+
+  // We don't expect test_step_cnt to 256 (flash mem is filled with all zeros)
+  // for this test. If it is 256, we just initialize the counters/NVM-fields
+  // again.
+  if (test_step_cnt == 256) {
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestSteps);
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestPhase);
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    rnd_num_iterations = flash_ctrl_testutils_counter_get(kCounterTestPhase);
+  }
+
+  // If this is the first iteration,
+  // set num_iterations to a random value between 8 and 32
+  if (rnd_num_iterations == 0) {
+    // 8 <= num_iters <= 32
+    rnd_num_iterations = rand_testutils_gen32_range(8, 32);
+    flash_ctrl_testutils_counter_set_at_least(&flash_ctrl, kCounterTestPhase,
+                                              rnd_num_iterations);
+    // LOG_INFO("phase1-num-iters = %d", rnd_num_iterations);
+  }
+
+  // Test phase #1: ping-timeout = 256
+  // No ping_timeout_fail alert shall be fired
+  // To ensure that the ping mechanism won't send spurious failure.
+  while (test_step_cnt < rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/256);
+    }
+
+    // Power-on reset
+    if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+      LOG_INFO("POR reset");
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      // Go to deep sleep mode
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+    } else { /*wakeup reset*/
+      // Check the wakeup reason and do the cleanup
+      check_wakeup_reason();
+      cleanup_wakeup_src();
+
+      // After the wakeup, wait enough, and
+      // check that no ping-timeout-fail alert is fired
+      wait_enough_for_alert_ping();
+      CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+          &alert_handler, kDifAlertHandlerLocalAlertAlertPingFail, &is_cause));
+      CHECK(!is_cause,
+            "Expected response is ping_timeout_fail= 0! But we got "
+            "ping_timeout_fail = 1");
```

Comment:
We should also check here that NONE of the other alerts (local or non-local) have fired when going through these transitions. We basically want to make sure that the LPG toggles do not produce spurious alert events.